### PR TITLE
Check functions against stdlib

### PR DESCRIPTION
**Pull request by nibalizer**
We don't want to duplicate a function in stdlib. Check for that.

### CODE REVIEW

File: spec/no_duplicate.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+
+git clone https://github.com/puppetlabs/puppetlabs-stdlib
+
+stdlib_funcs=`ls puppetlabs-stdlib/lib/puppet/parser/functions`
+extlib_funcs=`ls lib/puppet/parser/functions`
```
Comment (igalic) [inline spec/no_duplicate.sh]: [no](http://blag.esotericsystems.at/igor/says/no)
don't do that. http://mywiki.wooledge.org/ParsingLs

Comment (nibalizer) [inline spec/no_duplicate.sh]: These files are known to be ruby files... they wont have spaces or newlines or any of that jazz.

Comment (ffrank) [inline spec/no_duplicate.sh]: I'm with @igalic on this one. You're looping over the strings anyway, might as well call `basename` then.

File: spec/no_duplicate.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+for func in $extlib_funcs
+do
+    if `echo $stdlib_funcs | grep $func >/dev/null`; then
```
Comment (ffrank) [inline spec/no_duplicate.sh]: This pattern is too generic. You're doing but a substring match here, where you want exact matches, no?

Comment (nibalizer) [inline spec/no_duplicate.sh]: ah yes and basename makes that possbile. Neat, will respin

File: spec/no_duplicate.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  puts "Test failed"
+  exit 1
+else
+  puts "No duplicate functions found"
+  puts "Test passed"
+  exit 0
+end
+
+
+
```
Comment (igalic) [inline spec/no_duplicate.rb]: these lines left blank intentionally

**Comment by igalic** [comment]
can you add it to travis.yml? so we see it's actually working?
or rather, sync.yml?

File: spec/no_duplicate.rb (hunk 3 above)
Comment (nibalizer) [inline spec/no_duplicate.rb]: The script needs a pause to think about things.

**Comment by nibalizer** [comment]
@igalic it's been added to the :test target so yes travis ran this.

**Comment by igalic** [comment]
\o/ @nibalizer!

**Pull request merged**